evalml/automl/automl_search.py

```diff
@@ -908,16 +938,15 @@ def full_rankings(self):
 
     @property
     def best_pipeline(self):
-        """Returns an untrained instance of the best pipeline and parameters found during automl search.
+        """If `train_best_pipeline`, returns a trained instance of the best pipeline and parameters found during automl search. Otherwise, returns an untrained pipeline instance.
 
         Returns:
-            PipelineBase: untrained pipeline instance associated with the best automl search result.
+            PipelineBase: If `train_best_pipeline`, trained pipeline instance associated with the best automl search result. Otherwise, untrained pipeline instance.
         """
-        if not self.has_searched:
+        if not (self.has_searched and self._best_pipeline):
```

Comment:
Nice. Idk when the search will have been marked as ran but no best pipeline would get set, but seems fine to protect against that here.